### ISSUE TITLE
Timerfd_settime: Invalid argument, errno=22

### ISSUE THREAD

**Issue by simonxmh** [fluent/fluent-bit#1317]
## Bug Report

**Describe the bug**
I'm getting this error: 
```
[2019/05/06 13:18:39] [ Error] timerfd_settime: Invalid argument, errno=22 at /tmp/fluent-bit-1.0.5/lib/monkey/mk_core/mk_event_epoll.c:190 stream:stdout time:May 6th 2019, 09:18:39.028 

[2019/05/06 13:18:39] [ warn] [sched] retry for task 57 could not be scheduled stream:stderr time:May 6th 2019, 09:18:39.028
```
The node fd allocated is well below the maximum available fd, what exactly could be causing this error in that case?

**To Reproduce**
- Rubular link if applicable:
- Example log message if applicable:
```
[2019/05/06 13:18:39] [ Error] timerfd_settime: Invalid argument, errno=22 at /tmp/fluent-bit-1.0.5/lib/monkey/mk_core/mk_event_epoll.c:190 stream:stdout time:May 6th 2019, 09:18:39.028 

[2019/05/06 13:18:39] [ warn] [sched] retry for task 57 could not be scheduled stream:stderr time:May 6th 2019, 09:18:39.028
```
- Steps to reproduce the problem:
Using the fluentbit setup on a kubernetes cluster, exporting to an external elasticsearch endpoint with nginx-ingress as load balancer.

**Expected behavior**
That this error does not exist.

**Your Environment**

* Version used:1.0.5
* Configuration:
```
filter-kubernetes.conf: |
    [FILTER]
        Name                kubernetes
        Match               kube.*
        Kube_URL            https://kubernetes.default.svc:443
        Merge_Log           On
        Merge_Log_Key       k8s
        K8S-Logging.Parser  On
        K8S-Logging.Exclude On
  fluent-bit.conf: |
    [SERVICE]
        Flush         1
        Log_Level     info
        Daemon        off
        Parsers_File  parsers.conf
        HTTP_Server   On
        HTTP_Listen   0.0.0.0
        HTTP_Port     2020

    @INCLUDE input-kubernetes.conf
    @INCLUDE filter-kubernetes.conf
    @INCLUDE output-elasticsearch.conf
  input-kubernetes.conf: |
    [INPUT]
        Name              tail
        Tag               kube.*
        Path              /var/log/containers/*.log
        Parser            docker
        Docker_Mode       true
        Docker_Mode_Flush 4
        DB                /var/log/flb_kube.db
        Mem_Buf_Limit     10MB
        Skip_Long_Lines   true
        Rotate_Wait       30
        Refresh_Interval  10
  output-elasticsearch.conf: |
    [OUTPUT]
        Name            es
        Match           *
        Host            XXX
        Port            443
        Logstash_Format On
        Logstash_Prefix XXX
        Retry_Limit     False
        Generate_ID     true
        Buffer_Size     false
        Time_Key        fb-time
        tls on
        tls.verify true
        tls.debug 1
        tls.crt_file /secure/es-tls.crt
        tls.key_file /secure/es-tls.key
        tls.ca_file /secure/es-tls-ca.crt
  parsers.conf: |
    [PARSER]
        Name   json
        Format json
        Time_Key time
        Time_Format %Y-%m-%dT%H:%M:%S.%L

    [PARSER]
        Name        docker
        Format      json
        Time_Key    time
        Time_Format %Y-%m-%dT%H:%M:%S.%L
        Time_Keep   On
        Decode_Field_As   escaped   log
```

* Environment name and version (e.g. Kubernetes? What version?): Kubernetes v1.12.6-gke.10
* Server type and version:
* Operating System and version:
* Filters and plugins:

**Additional context**

This error has popped up at regular intervals, and I need to restart the service everytime, I hope to eliminate this issue.

**Comment by nvoskuilen** [comment]
Same experience with v1.0.6, suddenly not parsing logs.

`timerfd_settime: Invalid argument, errno=22 at /tmp/fluent-bit-1.0.6/lib/monkey/mk_core/mk_event_epoll.c:190`

Seems like more of this issue can be found here:
https://github.com/fluent/fluent-bit/issues/798
https://github.com/fluent/fluent-bit/issues/670
https://github.com/fluent/fluent-bit/issues/649

Running on DigitalOcean's k8s v1.13.3. Elasticsearch and fluent-bit reside in a cluster of 3 nodes.

```
[SERVICE]
    Flush        1
    Daemon       Off
    Log_Level    info
    Parsers_File parsers.conf

[INPUT]
    Name             tail
    Path             /var/log/containers/*.log
    Parser           docker
    Tag              kube.*
    Refresh_Interval 5
    Mem_Buf_Limit    5MB
    Skip_Long_Lines  On
    Docker_Mode       On

[FILTER]
    Name                kubernetes
    Match               kube.*
    Kube_URL            https://kubernetes.default.svc:443
    Kube_CA_File        /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
    Kube_Token_File     /var/run/secrets/kubernetes.io/serviceaccount/token

[OUTPUT]
    Name  es
    Match *
    Host  efk-elasticsearch
    Port  9200
    Logstash_Format On
    Retry_Limit False
    Type  flb_type
    Logstash_Prefix kubernetes_cluster
```

```
[PARSER]
    Name         docker
    Format       json
    Time_Key     time
    Time_Format  %Y-%m-%dT%H:%M:%S.%L
    Time_Keep    On
    # Command      |  Decoder | Field | Optional Action
    # =============|==================|=================
    Decode_Field_As   escaped_utf8    log    do_next
    Decode_Field_As   json            log
```

I used fluent-bit for the past 6 months in a bare-metal single node cluster setup without experiencing this issue. As I have understand, the only work-around for now is to redeploy fluent-bit every 2 hours.

**Comment by edsiper** [comment]
FYI: I've pushed a fix for the root cause of the problem, thanks to everybody who helped to troubleshoot the problem.

ref: https://github.com/fluent/fluent-bit/commit/1ef74fb0674490f665c3139a406896453f4d4a53

**Comment by edsiper** [comment]
Fixed in v1.1.1 release 

https://fluentbit.io/announcements/v1.1.1/
